### PR TITLE
(renderers) Allows renderers to be pluggable through modV.use

### LINKED ISSUE #262: Module types should be pluggable
For extensibility, modV Module types should be pluggable.

This should be relatively straightforward as there's a lot of repeated code in draw.js
https://github.com/m-o-d-v/modV/blob/2.0/src/modv/draw.js#L61

Most of this can be refactored and a standard "Render Context" in the form of: 
```JavaScript
{
  Module,
  canvas,
  context,
  video: modV.videoStream,
  features: state.audioFeatures,
  meyda,
  delta,
  pipeline,
}
```

can be passed to a renderer.

Along with this, the draw loop can further be refactored by normalising the pipeline process.

Something along the lines of this:

```JavaScript
function renderRenderer({
  canvas,
  context,
  renderFunction,
  renderContext,
  pipeline = false,
  layer,
}) {
  const { canvas: bufferCanvas, context: bufferContext } = getBufferCanvas();
  if (pipeline) {
    context.clearRect(0, 0, canvas.width, canvas.height);
    context.drawImage(
      bufferCanvas,
      0,
      0,
      canvas.width,
      canvas.height,
    );

    renderFunction(renderContext);

    bufferContext.clearRect(0, 0, canvas.width, canvas.height);
    bufferContext.drawImage(
      layer.canvas,
      0,
      0,
      canvas.width,
      canvas.height,
    );
  } else {
    renderFunction(renderContext);
  }
}
```

### PR DESCRIPTION
**Pull request by 2xAA**
Fixes #262.

### CODE REVIEW

**Comment by TimPietrusky** [comment]
@2xAA I wait for the review then until you tell me everything is fine!

**Comment by 2xAA** [comment]
@TimPietrusky This seems to be fine now, awaiting your review :)

File: src/modv/draw.js [hunk 1]
```diff
@@ -1,18 +1,61 @@
 import { modV } from '@/modv';
-import render2d from '@/modv/renderers/2d';
-import { render as renderShader } from '@/modv/renderers/shader';
-import { render as renderIsf } from '@/modv/renderers/isf';
+// import render2d from '@/modv/renderers/2d';
+// import { render as renderShader } from '@/modv/renderers/shader';
+// import { render as renderIsf } from '@/modv/renderers/isf';
```
Comment (TimPietrusky) [inline src/modv/draw.js]: Please remove unused imports.

**Review by TimPietrusky — CHANGES_REQUESTED** [review]
@2xAA Please remove unused imports.

File: src/modv/draw.js (hunk 1 above)
Comment (2xAA) [reply src/modv/draw.js]: @TimPietrusky This should be resolved now

**Comment by 2xAA** [comment]
@TimPietrusky Can we get this merged?

**Review by TimPietrusky — APPROVED** [review]

**Pull request merged**